### PR TITLE
Clean out opensmt

### PR DESCRIPTION
**Pull request by kroening**
- [ ] Each commit message has a non-empty body, explaining why the change was made.
- [x] My contribution is formatted in line with CODING_STANDARD.md.
- [ ] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- [ ] My commit message includes data points confirming performance improvements (if claimed).
- [ ] My PR is restricted to a single feature or bugfix.
- [x] White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

**Review by allredj — COMMENTED** [review]
Passed Diffblue compatibility checks (cbmc commit: 014eea49474c82b9345461da60ba925393797912).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/87877503

**Review by allredj — COMMENTED** [review]
Passed Diffblue compatibility checks (cbmc commit: 54cfd567820595916a317bf7402ae1cb6c6c805d).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/87881419

**Review by smowton — APPROVED** [review]
How hard would it be to add tests for the other solvers? That would enable testing the change to `arith_tools.cpp` for example?

**Comment by kroening** [comment]
The idea is that one day you can run all the tests in regression/cbmc with --cprover-smt2! We don't really need separate tests.

**Review by tautschnig — CHANGES_REQUESTED** [review]
Can we please not do "to_integer now converts uninterpreted bitvectors"? That completely contradicts what the comments say what a `bv_typet` is. I believe any interpretation of a `bv_typet` as a signed or unsigned bitvector should go via a type cast, bit- or byte extract operation. Or we just kill `bv_typet` completely and use `unsignedbv_typet` instead.

File: src/cbmc/cbmc_parse_options.h [hunk 1]
```diff
@@ -46,7 +46,8 @@ class optionst;
   OPT_GOTO_CHECK \
   "(no-assertions)(no-assumptions)" \
   "(xml-ui)(xml-interface)(json-ui)" \
-  "(smt1)(smt2)(fpa)(cvc3)(cvc4)(boolector)(yices)(z3)(opensmt)(mathsat)" \
+  "(smt1)(smt2)(fpa)" \
+  "(cprover-smt2)(cvc3)(cvc4)(boolector)(yices)(z3)(mathsat)" \
```
Comment (martin-cs) [inline src/cbmc/cbmc_parse_options.h]: OpenSMT is still being developed by Hana and Natasha's teams; do we really want to remove support?

**Comment by martin-cs** [comment]
@smowton : systematic testing of the SMT solvers would be good but I think it might have to be as part of a larger campaign of "let's make SMT solvers the first choice for back-ends" and each time we've done the benchmarking on this ... it hasn't (yet) justified the effort.

File: src/cbmc/cbmc_parse_options.h (hunk 1 above)
Comment (kroening) [reply src/cbmc/cbmc_parse_options.h]: Yes; it does not support any theories that we use.

Comment (martin-cs) [reply src/cbmc/cbmc_parse_options.h]: OK.

**Review by martin-cs — APPROVED** [review]
Given the claim that it no longer supports the necessary theories; remove seems sensible.

**Comment by martin-cs** [comment]
@tautschnig : This PR no longer includes the changes you objected to.

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: 145c895367aa5c6cf7d0080c277c731940a1c6d8).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/90441669

**Review by tautschnig — APPROVED** [review]

**Pull request merged**